go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -333,50 +353,24 @@ func (c *Controller) syncServices(key string) error {
 
 			// Node Port
 			if svcPort.NodePort != 0 {
-				gatewayRouters, _, err := gateway.GetOvnGateways()
+				if c.needsOVNLBUpdate(eps.IPs, service) {
+					if err := createPerNodeVIPs(nil, svcPort.Protocol, svcPort.NodePort, eps.IPs, eps.Port); err != nil {
```

Comment:
We are doing this for each ip in cluster ip, without checking the ip family.
Basically, adding the node ips twice if the cluster is dual stack.
For egress, this is addressed checking the ip family, not sure it would be better in a separate loop.

Reply:
good catch. Will fix this